prelude/slang-cuda-prelude.h

```diff
@@ -823,46 +828,121 @@ __inline__ __device__ T _waveReadLaneAtMultiple(T inVal, int lane)
     return outVal;
 }
 
-__device__ int _wavePrefixSum(int val)
+// Scalar 
+
+// Invertable means that when we get to the end of the reduce, we can remove val (to make exclusive), using 
+// the inverse of the op.
```

Comment:
Inverting the operation to back out the value for the current lane seems really undesirable, and could be incorrect in many cases. What if you are doing a product and the value is zero? What if you are doing a sum of floating-point values and the value is not finite (or *is* finite but causes the final result to be non-finite).

It seems like we need an approach that works for any "exclusive" op, including ones where invertability isn't really a thing, and at that point we should probably use that approach for all the ops (even ones where intertability might be usable).